amd_openvx_extensions/amd_rpp/source/ResizeMirrorNormalizeTensor.cpp

```diff
@@ -0,0 +1,337 @@
+	STATUS_ERROR_CHECK(vxQueryScalar((vx_scalar)parameters[10], VX_SCALAR_TYPE, &scalar_type, sizeof(scalar_type)));
+ 	if(scalar_type != VX_TYPE_UINT32) return ERRMSG(VX_ERROR_INVALID_TYPE, "validate: Paramter: #10 type=%d (must be size)\n", scalar_type);
+	STATUS_ERROR_CHECK(vxQueryScalar((vx_scalar)parameters[11], VX_SCALAR_TYPE, &scalar_type, sizeof(scalar_type)));
+ 	if(scalar_type != VX_TYPE_UINT32) return ERRMSG(VX_ERROR_INVALID_TYPE, "validate: Paramter: #11 type=%d (must be size)\n", scalar_type);
+	// Check for input parameters
+	vx_parameter input_param;
+	vx_image input;
+	vx_df_image df_image;
+	input_param = vxGetParameterByIndex(node,0);
+	STATUS_ERROR_CHECK(vxQueryParameter(input_param, VX_PARAMETER_ATTRIBUTE_REF, &input, sizeof(vx_image)));
+	STATUS_ERROR_CHECK(vxQueryImage(input, VX_IMAGE_ATTRIBUTE_FORMAT, &df_image, sizeof(df_image)));
+	if(df_image != VX_DF_IMAGE_U8 && df_image != VX_DF_IMAGE_RGB) {
+		return ERRMSG(VX_ERROR_INVALID_FORMAT, "validate: ResizeMirrorNormalizeTensor: image: #0 format=%4.4s (must be RGB2 or U008)\n", (char *)&df_image);
+	}
+
+	// Check for output parameters
+	vx_image output;
+	vx_df_image format;
+	vx_parameter output_param;
+	vx_uint32 height, width;
+	output_param = vxGetParameterByIndex(node, 3);
+	STATUS_ERROR_CHECK(vxQueryParameter(output_param, VX_PARAMETER_ATTRIBUTE_REF, &output, sizeof(vx_image)));
+	STATUS_ERROR_CHECK(vxQueryImage(output, VX_IMAGE_ATTRIBUTE_WIDTH, &width, sizeof(width)));
+	STATUS_ERROR_CHECK(vxQueryImage(output, VX_IMAGE_ATTRIBUTE_HEIGHT, &height, sizeof(height)));
+	STATUS_ERROR_CHECK(vxSetMetaFormatAttribute(metas[3], VX_IMAGE_ATTRIBUTE_WIDTH, &width, sizeof(width)));
+	STATUS_ERROR_CHECK(vxSetMetaFormatAttribute(metas[3], VX_IMAGE_ATTRIBUTE_HEIGHT, &height, sizeof(height)));
+	STATUS_ERROR_CHECK(vxSetMetaFormatAttribute(metas[3], VX_IMAGE_ATTRIBUTE_FORMAT, &df_image, sizeof(df_image)));
+	vxReleaseImage(&input);
+	vxReleaseImage(&output);
+	vxReleaseParameter(&output_param);
+	vxReleaseParameter(&input_param);
+	return status;
+}
+
+static vx_status VX_CALLBACK processResizeMirrorNormalizeTensor(vx_node node, const vx_reference * parameters, vx_uint32 num)
+{
+	RppStatus status = RPP_SUCCESS;
+	vx_status return_status = VX_SUCCESS;
+	ResizeMirrorNormalizeTensorLocalData *data = NULL;
+	STATUS_ERROR_CHECK(vxQueryNode(node, VX_NODE_LOCAL_DATA_PTR, &data, sizeof(data)));
+	vx_df_image df_image = VX_DF_IMAGE_VIRT;
+	STATUS_ERROR_CHECK(vxQueryImage((vx_image)parameters[0], VX_IMAGE_ATTRIBUTE_FORMAT, &df_image, sizeof(df_image)));
+
+#if ENABLE_HIP
+	if (data->device_type == AGO_TARGET_AFFINITY_GPU) {
+        refreshResizeMirrorNormalizeTensor(node, parameters, num, data);
+        status = rppt_resize_mirror_normalize_gpu((void *)data->hip_pSrc, data->srcDescPtr, (void *)data->hip_pDst, data->dstDescPtr, data->d_dstImgSize, RpptInterpolationType::BILINEAR, data->mean, data->std_dev, data->mirror, data->d_roiTensorPtrSrc, data->roiType, data->rppHandle);
```

Comment:
use static_cast() instead to avoid codacy error

Reply:
Used static_cast to fix the codacy warnings in the new commit